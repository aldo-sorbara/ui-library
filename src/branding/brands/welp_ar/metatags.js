export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
