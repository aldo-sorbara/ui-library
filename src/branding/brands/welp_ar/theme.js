import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
