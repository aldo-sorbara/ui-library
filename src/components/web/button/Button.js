import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * @render react
 * @name Button
 * @description Button Component for Wenance.
 * @example
 * <ButtonContainer>
 *   <Button data="button"> Button</Button>
 * </ButtonContainer>
 */

export const Button = props => {
  const { children, data, href } = props;
  const Component = href ? StyledLink : StyledButton;
  return (
    <Component {...props} data-w={data}>
      {children}
    </Component>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  data: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  isBlocked: PropTypes.bool,
  modifiers: PropTypes.arrayOf(PropTypes.string)
};

Button.defaultProps = {
  disabled: false,
  isBlocked: false,
  modifiers: []
};

const CommonButton = props => `
  border-radius: ${props.theme.buttons.borderRadius};
  display: ${props.isBlocked ? 'block' : 'inline-block'};
  cursor: pointer;
  font-family: inherit;
  opacity: ${props.disabled ? '0.3' : '1'};
  padding: ${props.theme.spacer.medium} ${props.theme.spacer.wrapperInner};
  text-align: center;
  transition: background-color 0.4s ease-in-out;
  touch-action: manipulation;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: ${props.isBlocked ? '100%' : 'auto'};
  cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
  background-color: ${props.theme.palette.primary.default};
  border: 1px solid transparent;
  color: ${props.theme.palette.common.text.inverse};

  &:hover:enabled {
    background-color: ${props.theme.palette.primary.dark};
  }

  &:focus {
    outline: none;
  }

  &:disabled,
  button[disabled] {
    opacity: 0.3;
  }
`;

CommonButton.defaultProps = {
  theme: {
    palette: {
      common: {
        text: {
          inverse: '#ffffff'
        }
      },
      primary: {
        default: '#0082ff',
        dark: '#3a66ab'
      }
    },
    typography: {
      button: {
        fontSize: '.875rem',
        fontWeight: '700',
        lineHeight: '1.145'
      }
    },
    spacer: {
      medium: '1rem',
      large: '1.5rem',
      button: {
        smallSpacer: '.75rem'
      }
    },
    buttons: {
      borderRadius: '0'
    }
  }
};

const StyledButton = styled.button`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
`;

const StyledLinkButton = props => `
  background: transparent;
  border: 1px solid transparent;
  color: ${props.theme.palette.primary.default};
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }`;

const StyledLink = styled.a`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
  text-decoration: none;
`;

const StyledSmallButton = props => `
  padding: ${props.theme.spacer.button.smallSpacer} ${props.theme.spacer.large};
`;

const StyledInverseButton = props => `
  background: transparent;
  color: ${props.theme.palette.common.text.inverse};
  border: 1px solid ${props.theme.palette.common.text.inverse};
  
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }
`;

const StyledRoundedButton = () => `
  border-radius: 50%;
  width: 50px;
  height: 50px
  padding: 0;

  svg {
    display: block;
    margin: auto;
  }
`;

const modifiers = {
  small: StyledSmallButton,
  rounded: StyledRoundedButton,
  link: StyledLinkButton,
  inverse: StyledInverseButton
};

// ButtonContainer Component
export const ButtonContainer = props => {
  return (
    <StyledButtonContainer {...props}>
      <div>{props.children}</div>
    </StyledButtonContainer>
  );
};

const StyledButtonContainer = styled.div`
  padding: ${props => props.theme.buttons.padding};
`;

StyledButtonContainer.defaultProps = {
  theme: {
    buttons: {
      padding: '0 2rem 2rem'
    }
  }
};
