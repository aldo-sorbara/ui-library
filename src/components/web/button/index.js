export { Button, ButtonContainer } from './Button';
