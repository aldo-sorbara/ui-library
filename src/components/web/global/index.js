export {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from './Global';
