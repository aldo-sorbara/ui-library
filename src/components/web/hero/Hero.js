import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

/**
 * @render react
 * @name Hero
 * @description Hero Component for Wenance
 */

const Hero = props => {
  const { children, imageUrl, title, titleStyle, action, ...rest } = props;
  const Component = StyledHero;
  return (
    <Component imageUrl={imageUrl} {...rest}>
      {title && <Title style={titleStyle}>{title}</Title>}
      {children && <DescriptionContainer>{children}</DescriptionContainer>}
      {action && <ActionContainer>{action}</ActionContainer>}
    </Component>
  );
};

Hero.propTypes = {
  children: PropTypes.node,
  action: PropTypes.node,
  imageUrl: PropTypes.string,
  title: PropTypes.string,
  titleStyle: PropTypes.object
};

Hero.defaultProps = {
  action: null,
  children: null,
  imageUrl: null,
  title: null,
  titleStyle: {}
};

const StyledHero = styled.div`
  background-image: url(${props =>
    props.imageUrl || props.theme.assetsPaths.landingBackground});
  background-position: 50% 0;
  background-repeat: no-repeat;
  background-size: cover;
  color: ${props => props.theme.palette.common.text.inverse};
  padding: ${props =>
    `${props.theme.spacer.hero.top} ${props.theme.spacer.wrapperInner} ${
      props.theme.spacer.hero.bottom
    }`};
  text-align: center;
`;

/* stylelint-disable order/properties-alphabetical-order */
const Title = styled.h1`
  ${props => props.theme.typography.headline};
  color: inherit;
  font-size: 2.25rem;
  margin-bottom: ${props => props.theme.spacer.wrapperInner};
`;
/* stylelint-enable order/properties-alphabetical-order */

const DescriptionContainer = styled.div`
  display: inline-block;
  margin-bottom: ${props => props.theme.spacer.large};
  text-align: center;
`;

const ActionContainer = styled.div`
  margin: 0 4rem;
`;

StyledHero.defaultProps = {
  theme: {
    spacer: {
      wrapperInner: '2rem',
      medium: '1rem',
      large: '1.5rem',
      hero: {
        bottom: '8rem',
        top: '13rem'
      }
    },
    palette: {
      common: {
        text: {
          inverse: '#ffffff'
        }
      }
    },
    assetsPaths: {
      landingBackground: './welp_ar/bg_welp.png'
    },
    typography: {
      headline: {
        'line-height': '1.14',
        'font-weight': '700',
        'font-size': '1.625rem'
      }
    }
  }
};

export default Hero;
