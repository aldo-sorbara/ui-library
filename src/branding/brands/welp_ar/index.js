import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
