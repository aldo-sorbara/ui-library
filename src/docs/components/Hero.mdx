---
name: Hero
menu: Components
route: /hero
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import Hero from '../../components/web/hero';
import { Button } from '../../components/web/button';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { styles } from '../../../utils/docz';

# Hero

<PropsTable of={Hero} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <Hero
        title='Soña en grande'
        style={{ backgroundColor: '#0082FF' }}
        action={
          <Button
            modifiers={['inverse']}
            handleOnClick={() => console.log('CLICKED')}
            data='siguente'
          >
            Comenzar
          </Button>
        }
      >
        Créditos personales 100% online, con plazos y tasas a tu medida.
      </Hero>
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/hero--with-title)
