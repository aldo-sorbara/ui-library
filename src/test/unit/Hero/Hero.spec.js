import React from 'react';
import Hero from '../../../components/web/hero';
import { Button } from '../../../components/web/button';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Hero component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Hero className='test-card' title='Soña en grande'>
          Hero children
        </Hero>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render hero with action prop', () => {
    const wrapper = mountWithTheme(
      <Hero
        className='test-card'
        title='Soña en grande'
        action={
          <Button
            modifiers={['inverse']}
            handleOnClick={() => console.log('CLICKED')}
            data='siguente'
          >
            Comenzar
          </Button>
        }
      >
        Hero children
      </Hero>,
      welp_ar.theme
    );
    expect(wrapper.props().action).not.toBe(null);
  });
  it('should render hero with custom background image prop', () => {
    const wrapper = mountWithTheme(
      <Hero
        className='test-card'
        title='Soña en grande'
        imageUrl='src/test'
        action={
          <Button
            modifiers={['inverse']}
            handleOnClick={() => console.log('CLICKED')}
            data='siguente'
          >
            Comenzar
          </Button>
        }
      >
        Hero children
      </Hero>,
      welp_ar.theme
    );
    expect(wrapper.props().imageUrl).toBe('src/test');
    expect(wrapper).toHaveStyleRule('background-image', 'url(src/test)');
  });
});
