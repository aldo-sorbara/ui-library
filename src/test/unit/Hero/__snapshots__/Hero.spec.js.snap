// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should render snapshot 1`] = `
.c0 {
  background-image: url(./public/welp_ar/bg_welp.png);
  background-position: 50% 0;
  background-repeat: no-repeat;
  background-size: cover;
  color: #ffffff;
  padding: 13rem 2rem 8rem;
  text-align: center;
}

.c1 {
  font-size: 1.625rem;
  font-weight: 700;
  line-height: 1.14;
  color: inherit;
  font-size: 2.25rem;
  margin-bottom: 2rem;
}

.c2 {
  display: inline-block;
  margin-bottom: 1.5rem;
  text-align: center;
}

<div
  className="test-card c0"
>
  <h1
    className="c1"
    style={Object {}}
  >
    So√±a en grande
  </h1>
  <div
    className="c2"
  >
    Hero children
  </div>
</div>
`;
